src/Tribe/Main.php

```diff
@@ -566,44 +566,52 @@ public function should_load_common_admin_css() {
-	 * First it tries to load the wp-content/languages translation then if falls to the try to load $dir language files.
+	 * A helper method to load text domain.
+	 * First, it tries to load language files from a custom folder when set. If it fails it falls back to load from $dir.
 	 *
 	 * @since  4.0.1 Introduced.
 	 * @since  4.2   Included $domain and $dir params.
+	 * @since  TBD   Fix text domain loading logic after changes in WordPress 6.7.
 	 *
 	 * @param string       $domain The text domain that will be loaded.
 	 * @param string|false $dir    What directory should be used to try to load if the default doesn't work.
 	 *
 	 * @return bool  If it was able to load the text domain.
 	 */
 	public function load_text_domain( $domain, $dir = false ) {
-		// Added safety just in case this runs twice...
+		// Added safety just in case this runs twice.
 		if ( is_textdomain_loaded( $domain ) && ! $GLOBALS['l10n'][ $domain ] instanceof NOOP_Translations ) {
 			return true;
 		}
 
-		$locale = get_locale();
-		$plugin_rel_path = WP_LANG_DIR . '/plugins/';
+		$locale          = get_locale();
+		$plugin_rel_path = $dir;
 
 		/**
-		 * Allows users to filter the file location for a given text domain..
-		 * Be careful when using this filter, it will apply across the whole plugin suite.
+		 * Allows users to filter the file location for a given text domain.
+		 * The path has to be relative to wp-content/plugins/.
+		 * Be careful when using this filter, it will apply across the whole plugin suite when not checking $domain.
 		 *
 		 * @param string      $plugin_rel_path The relative path for the language files.
-		 * @param string      $domain Which plugin domain we are trying to load.
-		 * @param string      $locale Which Language we will load.
-		 * @param string|bool $dir    If there was a custom directory passed on the method call.
+		 * @param string      $domain          The plugin domain we are trying to load.
+		 * @param string      $locale          The language we will load.
+		 * @param string|bool $dir             If there was a custom directory passed on the method call.
 		 */
 		$plugin_rel_path = apply_filters( 'tribe_load_text_domain', $plugin_rel_path, $domain, $locale, $dir );
 
-		$loaded = load_plugin_textdomain( $domain, false, $plugin_rel_path );
+		$filename       = $domain . '-' . $locale . '.mo';
+		$file           = trailingslashit( WP_PLUGIN_DIR ) . trailingslashit( $plugin_rel_path ) . $filename;
+		$mo_file_exists = file_exists( $file );
 
-		if ( $dir !== false && ! $loaded ) {
-			return load_plugin_textdomain( $domain, false, $dir );
+		// Load textdomain from a custom folder or the plugin's language folder.
+		if ( $mo_file_exists ) {
+			return load_plugin_textdomain( $domain, false, $plugin_rel_path );
 		}
 
-		return $loaded;
+		// If translation files are not found in the custom folder, then load textdomain from the plugin's language folder.
+		load_plugin_textdomain( $domain, false, $dir );
+
+		return true;
```

Comment:
Since `$domain` is being passed in let's utilize `load_plugin_textdomain` to determine whether to return true/false.
```suggestion
		// If translation files are not found in the custom folder, then load textdomain from the plugin's language folder.
		return load_plugin_textdomain( $domain, false, $dir );
```

Reply:
I do agree with you. However, when I was working on the test `load_plugin_textdomain` was returning false in certain scenarios.

Since this is the fallback, we could potentially either add an additional check to make sure that the fallback should work properly, or always return true.

I'm open for either option. The additional check could be checking for the `.mo` file in the `WP_LANG_DIR/plugins/*` directory. If it exists, we return true, if it doesn't we return false.